### PR TITLE
refactor: revamp flush policy module unit test cases

### PR DESCRIPTION
**Pull request by SKannaniOS**
## Description
This PR refactors the flush policy unit tests by migrating from the legacy XCTest framework to the modern Swift Testing framework. The changes break down the monolithic `FlushPolicyModuleTests.swift` file into separate, focused test files for each flush policy type and reorganize them into a proper module structure. This migration improves test organization, readability, and maintainability while adopting modern Swift testing practices.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Code refactor/optimization

## Implementation Details
- **Removed** the monolithic `FlushPolicyModuleTests.swift` file (253 lines) containing all flush policy tests
- **Created** dedicated test files using Swift Testing framework:
  - `CountFlushPolicyTests.swift` - Tests for count-based flush policies (165 lines)
  - `FrequencyFlushPolicyTests.swift` - Tests for frequency-based flush policies (178 lines) 
  - `StartupFlushPolicyTests.swift` - Tests for startup flush policies (62 lines)
  - `FlushPolicyFacadeTests.swift` - Tests for flush policy coordination (134 lines)
- **Reorganized** test structure into `PolicyModuleTests/FlushPolicies/` directory for better organization
- **Migrated** all test assertions from XCTest (`XCTAssert*`) to Swift Testing (`#expect`)
- **Enhanced** test coverage with more comprehensive test cases using parameterized testing
- **Moved** `MockAnalytics` class from `MockProvider.swift` to `SwiftTestMockProvider.swift` for better organization
- **Updated** Xcode project configuration to include new test files and exclude the old monolithic file

### Key Improvements:
- Better separation of concerns with dedicated test files for each policy type
- More descriptive test names following Swift Testing conventions
- Enhanced test coverage with parameterized tests
- Improved test readability using modern Swift Testing syntax
- Better test organization with logical grouping in directory structure

## Checklist
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation (if appropriate).
- [x] I have ensured that my code follows the project's code style.
- [x] I have checked for potential performance impacts and optimized if necessary.
- [x] I have checked the code for security issues.
- [ ] I have updated the changelog (if required).

## How to test?
The refactored tests can be verified by running the test suite:

1. **Run all tests**: Execute the test suite to ensure all flush policy tests pass
2. **Individual test execution**: Run each test file separately to verify isolation:
   - `CountFlushPolicyTests.swift` - Tests count-based flushing logic
   - `FrequencyFlushPolicyTests.swift` - Tests timer-based flushing logic  
   - `StartupFlushPolicyTests.swift` - Tests startup flushing behavior
   - `FlushPolicyFacadeTests.swift` - Tests policy coordination and management
3. **Test coverage verification**: Ensure test coverage is maintained or improved compared to the original tests
4. **Performance testing**: Verify that the new test structure doesn't impact test execution time

### Expected Results:
- All existing test functionality should be preserved
- Tests should run faster due to better organization
- Test output should be more descriptive with Swift Testing's enhanced reporting
- No functional changes to the actual flush policy implementations

## Breaking Changes
This PR does not introduce any breaking changes to the public API. The refactoring is purely internal to the test suite and does not affect the behavior of flush policies or any consumer-facing functionality.

## Maintainers Checklist
- [ ] The code has been reviewed.
- [ ] CI tests have passed.
- [ ] All necessary documentation has been updated.

## Screenshots (if applicable)
Not applicable for this refactoring.

## Additional Context
This refactoring is part of the ongoing effort to modernize the test suite and improve code organization across the RudderStack Swift SDK. The changes align with:

- **Modern Swift Testing Adoption**: Leveraging the latest Swift Testing framework features
- **Test Organization Best Practices**: Separating concerns and creating focused test modules
- **Improved Maintainability**: Making tests easier to understand, modify, and extend
- **Enhanced Developer Experience**: Better test discovery and execution with clearer naming

**Migration Notes**: 
- All original test functionality has been preserved
- Test coverage has been enhanced with additional edge cases
- Mock objects have been reorganized for better reusability
- No changes required for developers using the flush policy APIs

### CODE REVIEW

File: RudderStackAnalytics.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1258,10 +1284,12 @@
 				53DEA6722E03D06F00D6C371 /* RudderStackAnalyticsTests.swift in Sources */,
 				5393019E2E65562100DAEAD1 /* BackoffPolicyTests.swift in Sources */,
 				537B96482EABF7A100E3ADE7 /* SourceConfigTests.swift in Sources */,
-				859CEE162E9CDE9B00519C90 /* IntegrationPluginTests.swift in Sources */,
+				53E777582EB4C2BD00564D61 /* CountFlushPolicyTests.swift in Sources */,
+				859CEE162E9CDE9B00519C90 /* StandardIntegrationPluginTests.swift in Sources */,
 				859CEE162E9CDE9B00519C90 /* StandardIntegrationPluginTests.swift in Sources */,
```
[the remark this answers is not in the document]
Comment (SKannaniOS) [reply RudderStackAnalytics.xcodeproj/project.pbxproj:1289]: It'll be fixed in next PR.

**Review by 1abhishekpandey — APPROVED** [review]
